challenges/data.js

```diff
@@ -0,0 +1,84 @@
+//backend challenge
+const express = require('express');
+const router = express.Router();
+var num = null;
+var arr = Array(5000);
+
+router.post('/data', (req, res, next)=>{
+    num =  req.body.numbers;
+    var i = 0, count=0;
+    var regulator = new RegExp('^\\d+$');
+    var fact = true;
+    try{
+        for(i in num){
+           
+            if(regulator.test(num[i])){
+            arr[i] = num[i];
+            count++;
+            }
+            else{
+                fact = false;
+                break;
+            }
+        }
+        if(500===count&&fact){
+            res.status(201).json({
+                Message: 'Numbers Accepted',
+            });
+            sortArray();
+        }
+        else if(count<500&&fact){
+            res.status(201).json({
+                Error: 'Number list is too small'

```

Comment:
This comment goes for each of the Error messages. 

If I were a consumer of this API and the only interface for it I had was these error responses, I wouldn't be able to troubleshoot why my request failed if I passed in an array of strings or a picture of a turkey. The API doesn't need to handle a picture of a turkey, but when the request fails, I would like to know exactly why. 

I would also suggest using correct HTTP status codes for invalid responses.